tests/app/submitter/test_convert_payload_0_0_3.py

```diff
@@ -496,3 +496,85 @@ def test_default_answers_not_present_when_not_answered(fake_questionnaire_store)
+        Location(section_id='section-1', block_id='list-collector'),
+        Location(section_id='section-2', block_id='summary'),
+    ]
+
+    answer_objects = [
+        {'answer_id': 'you-live-here', 'value': 'Yes'},
+        {'answer_id': 'first-name', 'value': '1', 'list_item_id': 'xJlKBy'},
+        {'answer_id': 'last-name', 'value': '1', 'list_item_id': 'xJlKBy'},
+        {'answer_id': 'first-name', 'value': '2', 'list_item_id': 'RfAGDc'},
+        {'answer_id': 'last-name', 'value': '2', 'list_item_id': 'RfAGDc'},
+        {'answer_id': 'anyone-else', 'value': 'No'},
+    ]
+
+    answers = AnswerStore(answer_objects)
+
+    list_store = ListStore(
+        existing_items=[
+            {
+                'name': 'people',
+                'items': ['xJlKBy', 'RfAGDc'],
+                'primary_person': 'xJlKBy',
+            }
+        ]
+    )
+
+    fake_questionnaire_store.answer_store = answers
+    fake_questionnaire_store.list_store = list_store
+
+    schema = load_schema('test_list_collector_primary_person')
+
+    output = convert_answers(schema, fake_questionnaire_store, routing_path)
+
+    data_dict = json.loads(json.dumps(output['data']['lists'], for_json=True))
+
+    assert data_dict[0]['name'] == 'people'
+    assert "xJlKBy" in data_dict[0]['items']
+    assert data_dict[0]['primary_person'] == 'xJlKBy'
+
+
+def test_primary_person_not_in_payload_when_not_answered(fake_questionnaire_store):
+    routing_path = [
+        Location(section_id='section-1', block_id='list-collector'),
+        Location(section_id='section-1', block_id='next-interstitial'),
+        Location(section_id='section-1', block_id='another-list-collector-block'),
+        Location(section_id='section-2', block_id='summary'),
+    ]
+
+    answer_objects = [
+        {'answer_id': 'first-name', 'value': '1', 'list_item_id': 'xJlKBy'},
+        {'answer_id': 'last-name', 'value': '1', 'list_item_id': 'xJlKBy'},
+        {'answer_id': 'first-name', 'value': '2', 'list_item_id': 'RfAGDc'},
+        {'answer_id': 'last-name', 'value': '2', 'list_item_id': 'RfAGDc'},
+        {'answer_id': 'anyone-else', 'value': 'No'},
+        {'answer_id': 'another-anyone-else', 'value': 'No'},
+        {'answer_id': 'extraneous-answer', 'value': 'Bad', 'list_item_id': '123'},
+    ]
+
+    answers = AnswerStore(answer_objects)
+
+    list_store = ListStore(
+        existing_items=[{'name': 'people', 'items': ['xJlKBy', 'RfAGDc']}]
+    )
+
+    fake_questionnaire_store.answer_store = answers
+    fake_questionnaire_store.list_store = list_store
+
+    schema = load_schema('test_list_collector')
+
+    output = convert_answers(schema, fake_questionnaire_store, routing_path)
+
+    data_dict = json.loads(json.dumps(output['data']['lists'], for_json=True))
+
+    keys = []
+    for _, lists_key in enumerate(data_dict[0]):
+        keys.append(lists_key)
+
+    assert 'primary_person' not in keys
```

Comment:
This could be simplified to:
```python
assert 'primary_person' not in data_dict[0]
```

Reply:
@ajmaddaford we've simplified this test now, thanks